source/framework/core/inc/TRestMesh.h

```diff
@@ -60,16 +60,16 @@ class TRestMesh : public TObject {
     Int_t fNumberOfGroups = 0;
 
     /// A vector storing the group ID of the corresponding nodes activated
-    std::vector<Int_t> nodeGroupID;
+    std::vector<Int_t> fNodeGroupID;
     /// A vector storing the X-dimension cell id
-    std::vector<Int_t> nodeX;
+    std::vector<Int_t> fNodeX;
     /// A vector storing the Y-dimension cell id
-    std::vector<Int_t> nodeY;
+    std::vector<Int_t> fNodeY;
     /// A vector storing the Z-dimension cell id
-    std::vector<Int_t> nodeZ;
+    std::vector<Int_t> fNodeZ;
 
     /// A vector storing the total energy inside the cell id
-    std::vector<Double_t> energy;
+    std::vector<Double_t> fEnergy;
```

Comment:
Don't quite understand this design, you define different vectors with different parameters and then you assign the same index?
I would suggest to make a struct or class container:
```
class MeshParams {
Int_t ID;
Int_t fNodeX;
Int_t fNodeY;
Int_t fNodeZ;
Int_t fEnergy;
}
```
Then you just have to create an `std::vector<MeshParams>`

Reply:
Yes, that would be better, but right now I have no time for that ...

Reply:
Thats not this PR about. There are many other parts in the code where it would be good to have those kind of improvements.

Reply:
If you see that it is clearly an improvement to do it using a structure. Then, probably the best is that you create a new PR with those upgrades.

Reply:
If is spread everywhere in the code, I think you can create an issue about that.

Reply:
Sure, but before the issue for the structure we should make the issue for the `TRestMesh` validation routines, so if someone touches something at `TRestMesh` we make sure it doesn't break. xD

Reply:
If the vectors `std::vector<Double_t> fEnergy;` or similar are saved in a TTree, I guess it would break something or at least would not be backward compatible...

Reply:
Yes, it is a big change, not sure how good that would be accepted by ROOT schema evolution. But for the moment, there is no class storing `TRestMesh` on disk. It is only used as an auxiliary object, by `TRestAxionMagneticField` and `TRestFastHitsToTrackProcess`.